src/components/Layout/NetworkSelect/NetworkSelect.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownProps,
+  OverlayContext,
+  showDocumentTransferMessage,
+} from "@govtechsg/tradetrust-ui-components";
+import React, { FunctionComponent, useContext } from "react";
+import styled from "@emotion/styled";
+import { ChainId, ChainInfoObject } from "../../../constants/chain-info";
+import { useProviderContext } from "../../../common/contexts/provider";
+import { getChainInfo } from "../../../common/utils/chain-utils";
+
+interface NetworkSelectViewProps {
+  onChange: (network: ChainInfoObject) => void;
+  currentChainId: ChainId | undefined;
+  networks: ChainInfoObject[];
+}
+
+interface NetworkSelectDropdownItemProps extends DropdownItemLabelProps {
+  onClick?: () => void;
+}
+
+interface DropdownItemLabelProps {
+  network: ChainInfoObject;
+  className?: string;
+  active?: boolean;
+}
+
+/**
+ * Dropdown control for the network selection
+ */
+const WrappedDropdown = (props: DropdownProps) => {
+  const { children, className, ...rest } = props;
+  return (
+    <div className={className}>
+      <Dropdown className="rounded-md py-1 pl-4 p-2 border border-gray-300 bg-white" {...rest}>
+        {children}
+      </Dropdown>
+    </div>
+  );
+};
+
+const StyledDropdown = styled(WrappedDropdown)`
+  display: inline-block;
+  min-width: 12.5rem;
+  font-size: 0.87rem;
```

Comment:
tw if possible

Reply:
if we do granular control = write own css + maintaining that, which generally i feel we should avoid. often these gets forgotten and become stray leftover styles that no one will dare to remove later on.

im ok with 1-off inline style say `<div style={foo} />` or styled components with only those attributes that are not found in what tw offer out of box, say pseudo selectors, custom background-image etc.

otherwise for spacing / width matters, can stick to closest to what tw utilities provides (this way we adhere to standardisation by tw)

or if really so fancy till we need to maintain our own set, then we should define the css class either in tw js config, or tailind.css

Reply:
Got it!